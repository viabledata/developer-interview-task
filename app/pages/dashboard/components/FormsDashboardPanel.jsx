import React from "react";
import {withRouter} from "react-router";
import AppConstants from "../../../common/AppConstants";

export class FormsDashboardPanel extends React.Component {


    procedures(e) {
        e.preventDefault();
        this.props.history.replace({
            pathname: AppConstants.DASHBOARD_PATH, state: {
                shiftPresent: this.props.hasActiveShift
            }
        });
    }

    render() {
        return <li className="__card govuk-grid-column-one-third" id="proceduresPanel" style={{marginBottom: '30px'}}>
            <a href={AppConstants.DASHBOARD_PATH} onClick={this.procedures.bind(this)} className="card__body" id="proceduresPageLink">
                <span className="govuk-!-font-size-36 govuk-!-font-weight-bold">Forms</span>
            </a>
            <div className="card__footer">
                <span className="govuk-!-font-size-19">Submit a form</span>
            </div>
        </li>
    }
}

export default withRouter(FormsDashboardPanel)
