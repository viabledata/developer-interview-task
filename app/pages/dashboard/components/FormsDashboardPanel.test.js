import React from 'react';
import { mount } from 'enzyme';
import configureStore from 'redux-mock-store';
import { createMemoryHistory } from 'history';
import { Router } from 'react-router-dom';
import { FormsDashboardPanel } from './FormsDashboardPanel';

describe('Procedures Dashboard Panel', () => {
  const mockStore = configureStore();
  let store;
  const initialState = {
    'procedures-page': {},
  };
  beforeEach(() => {
    store = mockStore(initialState);
  });
  it('renders forms dashboard panel', async () => {
    const props = {};
    const wrapper = await mount(<FormsDashboardPanel
      store={store}
      {...props}
    />);
    expect(wrapper).toMatchSnapshot();
  });
  it('navigates to forms page on click', async () => {
    const history = createMemoryHistory('/forms');

    const props = {
      history,
      hasActiveShift: true,
    };
    const wrapper = await mount(<Router history={history}>
      <FormsDashboardPanel
        store={store}
        {...props}
      />
                                </Router>);

    const proceduresPageLink = wrapper.find('#proceduresPageLink');
    expect(proceduresPageLink.exists()).toEqual(true);

    proceduresPageLink.simulate('click');
    expect(props.history.location.pathname).toEqual('/forms');
  });
});
