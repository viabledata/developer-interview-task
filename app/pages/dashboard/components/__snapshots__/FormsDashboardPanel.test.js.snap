// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Procedures Dashboard Panel renders forms dashboard panel 1`] = `
<FormsDashboardPanel
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <li
    className="__card govuk-grid-column-one-third"
    id="proceduresPanel"
    style={
      Object {
        "marginBottom": "30px",
      }
    }
  >
    <a
      className="card__body"
      href="/forms"
      id="proceduresPageLink"
      onClick={[Function]}
    >
      <span
        className="govuk-!-font-size-36 govuk-!-font-weight-bold"
      >
        Forms
      </span>
    </a>
    <div
      className="card__footer"
    >
      <span
        className="govuk-!-font-size-19"
      >
        Submit a form
      </span>
    </div>
  </li>
</FormsDashboardPanel>
`;
